Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnCancel.hs

```diff
@@ -0,0 +1,66 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.Beckn.FRFS.OnCancel where
+
+import qualified Domain.Types.FRFSTicket as DFRFSTicket
+import qualified Domain.Types.FRFSTicketBooking as Booking
+import qualified Domain.Types.FRFSTicketBooking as FTBooking
+import qualified Domain.Types.FRFSTicketBookingPayment as DTBP
+import Domain.Types.Merchant as Merchant
+import Environment
+import Kernel.Beam.Functions
+import Kernel.Prelude
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified Storage.CachedQueries.Merchant as QMerch
+import qualified Storage.Queries.FRFSTicket as QTicket
+import qualified Storage.Queries.FRFSTicketBooking as QTBooking
+import qualified Storage.Queries.FRFSTicketBookingPayment as QTBP
+
+data OnCancelOrderStatus = SOFT_CANCEL | CANCELLED
```

Comment:
This needs to be moved in common enums file

Reply:
I added this here because `SOFT_CANCEL` is there in both `OnCancelOrderStatus` and `CancellationType` so we can't add them as enum in same file